fireapp/lib/pages/settings/setting_view_model.dart

```diff
@@ -0,0 +1,25 @@
+import 'dart:async';
+
+import 'package:fireapp/domain/repository/authentication_repository.dart';
+import 'package:fireapp/domain/repository/notification_fcm_token_repository.dart';
+import 'package:fireapp/presentation/fireapp_view_model.dart';
+import 'package:fireapp/presentation/login/login_navigation.dart';
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:injectable/injectable.dart';
+
+import '../../domain/models/notification/fcm_token.dart';
+
+@injectable
+class SettingViewModel {
+  final AuthenticationRepository _authenticationRepository;
+  final NotificationFCMTokenRepository _fcmTokenRepository;
+  SettingViewModel(this._authenticationRepository, this._fcmTokenRepository);
+
+  Future<void> unregisterFcmToken() async {
+    var curToken = await FirebaseMessaging.instance.getToken();
+    if (curToken == null) {
+      throw Exception('FCM token cannot be null');
+    }
+    _fcmTokenRepository.unregisterToken(curToken);
```

Comment:
@TaiHaDev Yep it was what I suspected! Easy fix, easy mistake. You need to await this line. What is currently happening is that it starts running this function then immediately returns (i.e, doesn't wait for the call to finish). The next line in the function that calls this then deletes the auth token. By the time `unregisterToken` gets to making the call, the authentication details are no longer there. When you add an `await` before this, the function won't return until `unregisterToken` finishes.

Reply:
Got it! I have fixed it below. Thanks for the detailed explanation.